### ISSUE TITLE
Reconnect when websocket closed remotely (opcode=8, a.k.a. the random errorless disconnect problem)

### ISSUE THREAD

**Issue by Laogeodritt** [Worldbuilding/KazTron#10]
```
[2018-02-11 16:11:18,241] (DEBUG) websockets.protocol: client << Frame(fin=True, opcode=8, data=b'\x03\xe8') [in /home/marcalexc/.virtualenvs/kaztron/lib/python3.6/site-packages/websockets/protocol.py:548]
[2018-02-11 16:11:18,241] (DEBUG) websockets.protocol: client >> Frame(fin=True, opcode=8, data=b'\x03\xe8') [in /home/marcalexc/.virtualenvs/kaztron/lib/python3.6/site-packages/websockets/protocol.py:564]
[2018-02-11 16:11:18,244] (INFO) kaztron.bootstrap: Closing event loop... [in /home/marcalexc/data/sw/KazTron/KazTron.py:64]
[2018-02-11 16:11:18,245] (INFO) kaztron.bootstrap: Exiting. [in /home/marcalexc/data/sw/KazTron/KazTron.py:66]
```

No notable traffic prior to this - normal message send/receives, user status changes, at least 60s prior to this.

Appears to be the remote server closing the websocket?

**Comment by Laogeodritt** [comment]
```
[2018-02-11 16:11:18,241] (DEBUG) websockets.protocol: client << Frame(fin=True, opcode=8, data=b'\x03\xe8') [in /home/marcalexc/.virtualenvs/kaztron/lib/python3.6/site-packages/websockets/protocol.py:548]
[2018-02-11 16:11:18,241] (DEBUG) websockets.protocol: client >> Frame(fin=True, opcode=8, data=b'\x03\xe8') [in /home/marcalexc/.virtualenvs/kaztron/lib/python3.6/site-packages/websockets/protocol.py:564]
[2018-02-11 16:11:18,244] (INFO) kaztron.bootstrap: Closing event loop... [in /home/marcalexc/data/sw/KazTron/KazTron.py:64]
[2018-02-11 16:11:18,245] (INFO) kaztron.bootstrap: Exiting. [in /home/marcalexc/data/sw/KazTron/KazTron.py:66]
```

No notable traffic prior to this - normal message send/receives, user status changes, at least 60s prior to this.

Appears to be the remote server closing the websocket?

**Comment by Laogeodritt** [comment]
Attempt at top-level mitigation in 1605caaae46496b3e25ac95dff95fa7522d18116

**Comment by Laogeodritt** [comment]
Attempted resolution in commit 8461ad2f4594e123fabffb873dc3cb786742bacf - see #15

**Comment by Laogeodritt** [comment]
Merged in #15
